File: src/test/java/io/binac/leetcode/SearchInRotatedSortedArrayIITest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class SearchInRotatedSortedArrayIITest {
    private final SearchInRotatedSortedArrayII.Solution1 solution1 = new SearchInRotatedSortedArrayII.Solution1();

    @Test
    void test1() {
        int nums1[] = {2, 5, 6, 0, 0, 1, 2};
        assertTrue(solution1.search(nums1, 0));
        int nums2[] = {2, 5, 6, 0, 0, 1, 2};
        assertFalse(solution1.search(nums2, 3));
        int nums3[] = {1,1,3,1};
        assertTrue(solution1.search(nums3, 3));
    }
}

File: cpp/tests/SearchInRotatedSortedArrayIITest.cpp
#include "SearchInRotatedSortedArrayII.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(SearchInRotatedSortedArrayII, Solution_1) {
  auto S81_1 = Solution81_1();

  std::vector<int> Nums1{2, 5, 6, 0, 0, 1, 2};
  EXPECT_TRUE(S81_1.search(Nums1, 0));
  EXPECT_FALSE(S81_1.search(Nums1, 3));
}